src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -3120,3 +3216,20 @@ void Creature::ClearTextRepeatGroup(uint8 textGroup)
     if (groupItr != m_textRepeat.end())
         groupItr->second.clear();
 }
+
+bool Creature::IsEscortNPC(bool isEscorting)
+{
+    if (!GetAI())
+        return false;
+
+    if (npc_escortAI* escortAI = dynamic_cast<npc_escortAI*> (AI()))
```

Comment:
this need to be virtual method in CreatureAI*, do not create a dependency on ai type here